### PR TITLE
added debug info to OperatorDef

### PR DESCRIPTION
**Pull request by bwasti**
this might be useful for net construction debugging.

### CODE REVIEW

**Comment by xiaomengy** [comment]
@caffe2bot retest this please

**Comment by bddppq** [comment]
onnx has a similar field called [doc_string](https://github.com/onnx/onnx/blob/5641b4abd364d32710ec19d22f3ae321282fda9b/onnx/onnx.proto#L169). I think debug_info is a more appropriate name, but to align with onnx maybe we should use doc_string?

File: caffe2/python/core.py [hunk 1]
```diff
@@ -321,6 +322,12 @@ def CreateOperator(
     registered with Caffe2.
     """
     operator = caffe2_pb2.OperatorDef()
+    stack = traceback.extract_stack()
+    debug_info = ""
+    for frame in stack[:-1]:
+      debug_info += "{}:{} {} -> {}\n".format(*frame)
```
Comment (bddppq) [inline caffe2/python/core.py]: https://docs.python.org/2/library/traceback.html#traceback.format_exc?

**Comment by bwasti** [comment]
`doc_string` is a very confusing name for a stacktrace

**Review by bddppq — APPROVED** [review]

**Comment by bwasti** [comment]
fix is here :) https://github.com/caffe2/caffe2/pull/2013

**Pull request merged**